README.md

```diff
@@ -80,6 +81,16 @@ empty_array_mt
 --------------
 **syntax:** `setmetatable({}, cjson.empty_array_mt)`
 
+**NOTE**: Use of this field is *discouraged* as of the `2.1.0.6` release. This
+field is considered *depcrecated*, and has been renamed to
+[array_mt](#array_mt) for expliciteness.
+
+[Back to TOC](#table-of-contents)
+
+array_mt
+--------
+**syntax:** `setmetatable({}, cjson.array_mt)`
+
 A metatable which can "tag" a table as a JSON Array in case it is empty (that is, if the
```

Comment:
So the behavior we want from `cjson.array_mt` is slightly different than `cjson.empty_array_mt` then? Currently, `cjson.empty_array_mt` encodes a table as an Array if it is empty (avoids having to do the `if #t < 1` thing.

What it seems we want here is something that encodes as a JSON Array regardless of its content? It then seems both metatables belong to this module as they solve different use cases?